frontend/packages/dev-console/src/components/monitoring/alerts/MonitoringAlerts.tsx

```diff
@@ -94,41 +94,43 @@ export const MonitoringAlerts: React.FC<props> = ({ match, rules, filters, listS
     setRows(tableRows);
   }, [collapsedRowsIds, filteredRules, namespace]);
 
-  const onCollapse = (event: React.MouseEvent, rowKey: number, isOpen: boolean) => {
-    rows[rowKey].isOpen = isOpen;
-    const { id } = rows[rowKey].cells[0];
-    if (!_.includes(collapsedRowsIds, id)) {
-      setCollapsedRowsIds([...collapsedRowsIds, id]);
-    } else if (_.includes(collapsedRowsIds, id)) {
-      setCollapsedRowsIds(_.without(collapsedRowsIds, id));
-    }
-    setRows([...rows]);
-  };
-  const handleSort = (_event: React.MouseEvent, index: number, direction: SortByDirection) => {
-    dispatch(
-      sortList(
-        reduxID,
-        monitoringAlertColumn[index - 1].fieldName,
-        monitoringAlertColumn[index - 1].sortFunc,
-        direction,
-        monitoringAlertColumn[index - 1].title,
-      ),
-    );
-    setSortBy({ index, direction });
-  };
-
-  if (loading && !loadError) {
-    return <LoadingBox />;
-  }
-  if (_.isEmpty(response?.data?.groups)) {
-    return <EmptyBox label={t('devconsole~Alerts')} />;
-  }
+  const onCollapse = React.useCallback(
+    (event: React.MouseEvent, rowKey: number, isOpen: boolean) => {
+      rows[rowKey].isOpen = isOpen;
+      const { id } = rows[rowKey].cells[0];
+      if (!_.includes(collapsedRowsIds, id)) {
+        setCollapsedRowsIds([...collapsedRowsIds, id]);
+      } else if (_.includes(collapsedRowsIds, id)) {
+        setCollapsedRowsIds(_.without(collapsedRowsIds, id));
+      }
+      setRows([...rows]);
+    },
+    [collapsedRowsIds, rows],
+  );
+  const handleSort = React.useCallback(
+    (_event: React.MouseEvent, index: number, direction: SortByDirection) => {
+      dispatch(
+        sortList(
+          reduxID,
+          monitoringAlertColumn[index - 1].fieldName,
+          monitoringAlertColumn[index - 1].sortFunc,
+          direction,
+          monitoringAlertColumn[index - 1].title,
+        ),
+      );
+      setSortBy({ index, direction });
+    },
+    [dispatch, monitoringAlertColumn],
+  );
 
-  return (
-    <>
-      <Helmet>
-        <title>{t('devconsole~Alerts')}</title>
-      </Helmet>
+  const Content = React.useMemo(() => {
+    if (loading && !loadError) {
+      return <LoadingBox />;
+    }
+    if (_.isEmpty(response?.data?.groups)) {
+      return <EmptyBox label={t('devconsole~Alerts')} />;
+    }
+    return (
```

Comment:
Is this inadvertently added to fix another issue?

Reply:
No, I did this so that title will be set for Empty state and loading state as well.